python/packages/autogen-ext/src/autogen_ext/models/_openai/_openai_client.py

```diff
@@ -877,6 +936,57 @@ def __setstate__(self, state: Dict[str, Any]) -> None:
 
 
 class AzureOpenAIChatCompletionClient(BaseOpenAIChatCompletionClient):
+    """Chat completion client for Azure OpenAI hosted models.
+
+    Args:
+        azure_endpoint (str): The endpoint for the Azure model. **Required for Azure models.**
+        model (str): The deployment ID for the Azure model. **Required for Azure models.**
+        api_version (str): The API version to use. **Required for Azure models.**
+        azure_ad_token (str): The Azure AD token to use. Provide either this or `azure_ad_token_provider`.
+        azure_ad_token_provider (AsyncAzureAdTokenProvider): The Azure AD token provider to use. Provide either this or `azure_ad_token`.
+        model_capabilities (ModelCapabilities): The capabilities of the model. **Required for Azure models.**
+        api_key (optional, str): The API key to use, use this if you are using key based authentication.
```

Comment:
Should also support an environment variable `AZURE_OPENAI_API_KEY`

Reply:
Updated